src/ert/_c_wrappers/util/substitution_list.py

```diff
@@ -117,14 +118,13 @@ def substitute(self, to_substitute: str) -> str:
     def substitute_real_iter(
         self, to_substitute: str, realization: int, iteration: int
     ) -> str:
-        to_substitute = self._alloc_filtered_string(to_substitute)
+        copy_substituter = copy.deepcopy(self)
         geo_id_key = f"<GEO_ID_{realization}_{iteration}>"
-        to_substitute = to_substitute.replace(
-            "<GEO_ID>", self.get(geo_id_key, "<GEO_ID>")
-        )
-        to_substitute = to_substitute.replace("<IENS>", str(realization))
-        to_substitute = to_substitute.replace("<ITER>", str(iteration))
-        return to_substitute
+        if geo_id_key in self:
+            copy_substituter.addItem("<GEO_ID>", self[geo_id_key])
+        copy_substituter.addItem("<IENS>", str(realization))
+        copy_substituter.addItem("<ITER>", str(iteration))
+        return copy_substituter.substitute(to_substitute)
```

Comment:
Are these changes guarding against recursive defines like 
```
DEFINE <ITER> <ITER>
```
If not then it is not really needed to fix the arglist substitution.  And if they are then what about guarding against 
```
DEFINE <KEY> <KEY>

```
?

Reply:
I know that sounds like I am paranoid, but we keep discovering real world edge cases, so I thought it would be better to stick with the way of doing it that was more intuitive to the user. I would argue that it is not intuitive for the user that `<IENS>` has a special case were it can get starved from being chosen for substitution. All other keys are chosen fairly (all given one go per round of substitutions).